File: backend/backend/Areas/Main/Models/LeadNotes.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Areas.Main.Models;

public class LeadNotes
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public int LeadId { get; set; }
    public Lead Lead { get; set; }
    public int NoteId { get; set; }
    public Note Note { get; set; }
}

File: frontend/lib/models/lead_notes.dart
import 'package:frontend/models/lead.dart';
import 'package:frontend/models/note.dart';

class LeadNotes {
  int? id;
  Lead? lead;
  Note? note;
  int? leadId;
  int? noteId;

  LeadNotes({
    this.id,
    this.lead,
    this.note,
    this.leadId,
    this.noteId
  });

  factory LeadNotes.fromJson(Map<String, dynamic> json) {
    return LeadNotes(
      id: json['id'],
      leadId: json['leadId'],
      lead: json['lead'] != null ? Lead.fromJson(json['lead']) : null,
      noteId: json['noteId'],
      note: json['note'] != null ? Note.fromJson(json['note']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'lead': lead,
      'note': note,
      'noteId': noteId,
      'leadId': leadId
    };
  }
}